src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/TestClasses/ConstaintsTestExtensions.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Extensions.Logging.Generators.Tests.TestClasses
+{
+    using ConstraintInAnotherNamespace;
+    namespace UsesConstraintInAnotherNamespace
+    {
+        public partial class MessagePrinter<T>
+            where T : Message
+        {
+            public void Print(ILogger logger, T message)
+            {
+                Log.Message(logger, message.Text);
+            }
+
+            internal static partial class Log
```

Comment:
Would it make sense for the test to also have the constraint with a class in another namespace on the logger class itself?

Reply:
Added test